library/core/src/time.rs

```diff
@@ -1081,3 +1157,55 @@ impl fmt::Debug for Duration {
         }
     }
 }
+
+/// An error which can be returned when converting a floating-point value of seconds
+/// into a [`Duration`].
+///
+/// This error is used as the error type for [`Duration::try_from_secs_f32`] and
+/// [`Duration::try_from_secs_f64`].
+///
+/// # Example
+///
+/// ```
+/// #![feature(duration_checked_float)]
+///
+/// use std::time::Duration;
+///
+/// if let Err(e) = Duration::try_from_secs_f32(-1.0) {
+///     println!("Failed conversion to Duration: {}", e);
+/// }
+/// ```
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[unstable(feature = "duration_checked_float", issue = "83400")]
+pub struct FromSecsError {
+    kind: FromSecsErrorKind,
+}
+
+impl FromSecsError {
+    const fn description(&self) -> &'static str {
```

Comment:
`Error::description` is deprecated; please move this match to the implementation of `Display::fmt`.

Reply:
@joshtriplett 
This uses a separate `const` method so that it could be used for the panic message in the functions `Duration::from_secs_{f32, f64}`.

Reply:
I did see that, but what I missed is that those can't actually use the Display impl themselves.

Reply:
Ok. That's fine then